opennms-webapp/src/main/webapp/js/angular-onms-quicksearch.js

```diff
@@ -0,0 +1,532 @@
+/**
+ * Convert from a clause into a FIQL query string.
+ */
+function toFiql(clauses) {
+	var first = true;
+	var fiql = '';
+	for (var i = 0; i < clauses.length; i++) {
+		if (!first) {
+			// Use OR to combine all operations
+			fiql += ',';
+		}
+		fiql += clauses[i].property;
+
+		switch (clauses[i].operator) {
+		case 'EQ':
+			fiql += '=='; break;
+		case 'NE':
+			fiql += '!='; break;
+		case 'LT':
+			fiql += '=lt='; break;
+		case 'LE':
+			fiql += '=le='; break;
+		case 'GT':
+			fiql += '=gt='; break;
+		case 'GE':
+			fiql += '=ge='; break;
+		}
+
+		fiql += escapeSearchValue(clauses[i].value);
+
+		first = false;
+	}
+	return fiql;
+}
+
+/**
+ * Escape FIQL reserved characters by URL-encoding them. Reserved characters are:
+ * <ul>
+ * <li>!</li>
+ * <li>$</li>
+ * <li>'</li>
+ * <li>(</li>
+ * <li>)</li>
+ * <li>*</li>
+ * <li>+</li>
+ * <li>,</li>
+ * <li>;</li>
+ * <li>=</li>
+ * </ul>
+ * @param value
+ * @returns String with reserved characters URL-encoded
+ */
+function escapeSearchValue(value) {
+	if (typeof value === 'string') {
```

Comment:
Is there any reason you're not just doing the builtin JS `encodeURIComponent()` here instead?  Pretty sure this will be slower *and* we already know CXF is URL-decoding it on their side anyways...